CHECK_POWER_COMMANDS/Backstop_File_Class.py

```diff
@@ -5,13 +5,16 @@
 #                        backstop file complete with any error messages
 #                        generated by the Check_Power_cmds program.
 #
+# Update: February 22, 2021
+#         Gregg Germain
+#         - Fixed the erroneous firing of Rule #5 - the 63 second timing error
+#
 ################################################################################
+from astropy.time import Time, TimeCxcSec
```

Comment:
I woudld have done this:

```python
from cxotime import CxoTime
```
and then `CxoTime` is a drop-in replacement for `DateTime` (has the same methods and attributes).

Reply:
Indeed Ska tools should never use `Time` directly, only `CxoTime` is approved and tested for Chandra operations. 

Also the time format classes like `TimeCxcSec` are not normally used in application code, and this import appears to be unused in your code.

Reply:
TimeCxcSec import removed.

Check_Power_Cmds.py is not a "Ska Tool" any more than another tool, called right after,  by the same Perl script:  Window_Check.py.  Window_Check is executed with Python3 because that's all that's required: No Ska-functionality required.

Check_Power_Cmds.py is a small tool which needs to convert CR*.backstop Chandra DOY  to Chandra seconds.  Though not required, Chandra.Time was handy so it was originally used it for the conversion.  Chandra.Time is deprecated and so removed and there is no need for that, nor any other Ska-specific functionality in this tool.

Instead it now uses Astropy Time - **as does cxotime** - for the conversion. Astropy Time is just as handy as Chandra.Time was.

This makes the tool lighter, far more portable, and faster

### Lighter - no need to import huge amounts of unused capability

cxotime imports Astropy Time so is layered on top of Astropy Time.
 I use Astropy Time directly.

cxotime is a very capable package that can perform many conversions.:

   byear, byear_str, cxcsec, datetime, decimalyear, fits, gps, iso, isot, jd, jyear, jyear_str, mjd, plot_date , unix and yday with great and maude formats

This modified tool doesn't need any of those other formats nor any other Ska capability. 
There is simply no need to impose the extra layer on top of the package built for and used by the general Astronomical community (Astropy), and which the extra layer (cxotime), in the end, itself imports.

### More portable -  All that is required is Python3 and Astropy. 

The tool can be, and is, run in stand alone mode outside of the Production ACIS Ops load review process.

If Royce wants to use or experiment with additions or mods to the tool,  at home,  or Catherine (or anyone) wants to do the same  on an MIT machine, they can so long as they have Python3 and  Astropy installed (similar to Window_Check).

Astropy is not some deep python package that only python gurus know about. It's a top level package used by astronomers.

From the Astropy web page:

"The Astropy Project is a community effort to develop a **common core package for Astronomy in Python and foster an ecosystem of interoperable astronomy packages**. "


### Faster - No need to spend time importing Ska.  

Importing cxotime adds an extra  layer all in order to get to essentially the same Astropy call.  cxotime parses/guesses  - very handy capability but not needed here:

In this app the formats are fixed and known driven by the CR*.backstop file and the need
to convert DOY to seconds. That simple need does not require  the extra layer of cxotime.

Also from the Astropy Time web page:

The astropy.time package is based on fast and memory efficient PyERFA wrappers around the ERFA time and calendar routines.

 **However, if the format is known in advance the string parsing will be faster if the format is provided.**

Therefore there is no need or desire to impose Ska invocation on this particular tool. It impedes portability, it runs faster without it, the code is much lighter in weight, it uses the same tool used by cxotime to do the conversion, and uses it far more directly and efficiently,  will not require modification and/or re-test when Ska changes (code maintenance costs being - by far - the largest software cost),  and Astropy is well known and well used by the community.

Reply:
OK. Just be aware that `CxoTime` fixes a compatibility issue related to `cxcsec` that will catch many non-expert users by surprise.